src/lower.c

```diff
@@ -10534,15 +10534,27 @@ static vcode_unit_t lower_concurrent_block(tree_t block, vcode_unit_t context)
 
    vcode_unit_t vu = emit_instance(name, tree_to_object(block), context);
 
+   lower_push_scope(block);
+
    if (cover_enabled(cover_tags, COVER_MASK_ALL)) {
       if (!context)
          cover_reset_scope(cover_tags, prefix);
-
       cover_push_scope(cover_tags, block);
+
+      lower_scope_t *it;
+      for (it = top_scope; it != NULL; it = it->down) {
```

Comment:
This needs rebasing as I refactored a lot of this code (see `lower_instance`).

Reply:
OK, done. I have iterated to through the `decls` t find out the `T_HIER` that could lead me to the
entity name of the current instance. I needed to do this before `lower_decls` on the given instance,
since I wanted to to take into account toggle tags on ports (If I disable coverage collecting on an
entity, I don't want to collect coverage on ports of its instance).

I hope its not too much of duplication against `lower_decls` -> `lower_decl` -> `lower_inst`.